src/runtime/GlobalObjectBuiltinString.cpp

```diff
@@ -183,6 +183,25 @@ static Value builtinStringSubstr(ExecutionState& state, Value thisValue, size_t
 
 static Value builtinStringMatch(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
 {
+#ifdef ESCARGOT_ENABLE_ES2015
+    RESOLVE_THIS_BINDING_TO_OBJECT(S, String, match);
+    Value argument = argv[0];
+    RegExpObject* regexp;
+
+    if (argument.isUndefinedOrNull() == false) {
+        Value match = argument.asObject()->getMethod(state, argument, ObjectPropertyName(state, state.context()->vmInstance()->globalSymbols().match));
+        if (match.isUndefined() == false) {
+            return Object::call(state, match, argument, 1, &thisValue);
+        }
+    }
+
+    String* str = Value(S).toString(state);
```

Comment:
Calling the `toString` on S is unnecessary. (hint: RESOLVE_THIS_BINDING_TO_STRING)